File: 2000 98 & ME/QuickStart/howto/samples/XMLSerialization/VB/readwritexml.vb
Imports System.Xml.Serialization
Imports System.IO
Imports XmlSerializationHowTo

public Class Test
	Shared Sub Main()
        	Dim serializer as XmlSerializer
		serializer = new XmlSerializer(GetType(PurchaseOrder))

        	Dim reader as TextReader 
		reader = new StreamReader("..\PurchaseOrder.xml")

	        Dim po as PurchaseOrder 
		po = CType(serializer.Deserialize(reader), PurchaseOrder)

	        reader.Close

	        Dim writer as TextWriter 
		writer = new StreamWriter("PurchaseOrder2.xml")

        	serializer.Serialize(writer, po)
        	writer.Close()
	End Sub
End Class

File: 2000 98 & ME/QuickStart/howto/samples/XMLSerialization/CS/ReadWriteXML.cs
using System.Xml.Serialization;
using System.IO;
using XmlSerializationHowTo;

public class Test {
    public static void Main(string[] args) {
        XmlSerializer serializer = new XmlSerializer(typeof(PurchaseOrder));

        TextReader reader = new StreamReader("..\\PurchaseOrder.xml");
        PurchaseOrder po = (PurchaseOrder)serializer.Deserialize(reader);
        reader.Close();

        TextWriter writer = new StreamWriter("PurchaseOrder2.xml");
        serializer.Serialize(writer, po);
        writer.Close();
    }
}
